test/distributed/test_c10d_nccl.py

```diff
@@ -575,6 +575,33 @@ def test_set_nccl_pg_timeout(self, backend):
         c10d.distributed_c10d._set_pg_timeout(timedelta(seconds=252), pg)
         self._check_nccl_timeout(timedelta(seconds=252))
 
+    @requires_nccl()
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    @parametrize("backend", [None, "nccl"])
+    def test_extend_nccl_pg_timeout(self, backend):
```

Comment:
This test somehow will fail periodically. So I have L593 which will give me different address on the python side.

Reply:
The reason test failed is that, after we set time, somehow when we get the backend and Options again, it now points to a different address and the timeout value was mess up. It is 701 secs instead of 600..

Reply:
Solved.